File: app/src/main/java/com/github/windsekirun/yukarisynthesizer/notification/BackgroundActivity.java
package com.github.windsekirun.yukarisynthesizer.notification;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;


public class BackgroundActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        onNewIntent(getIntent());
    }

    @Override
    protected void onNewIntent(final Intent intent) {
        startApp(intent);
        finish();
    }

    public void startApp(Intent intent) {
        if (isTaskRoot()) {
            goIntro(this, intent);
        } else {
            goMain(this, intent);
        }
    }

    private static void goIntro(Context context, Intent intent) {
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
//        intent.setClass(context, MainActivity.class);
        context.startActivity(intent);
    }

    private static void goMain(Context context, Intent intent) {
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
//        intent.setClass(context, MainActivity.class);
        context.startActivity(intent);
    }
}

File: app/src/main/java/com/github/windsekirun/yukarisynthesizer/notification/BackgroundActivity.kt
package com.github.windsekirun.yukarisynthesizer.notification

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.github.windsekirun.yukarisynthesizer.main.MainActivity


class BackgroundActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        onNewIntent(intent)
    }

    override fun onNewIntent(intent: Intent) {
        startApp(intent)
        finish()
    }

    fun startApp(intent: Intent) {
        if (isTaskRoot) {
            goIntro(this, intent)
        } else {
            goMain(this, intent)
        }
    }

    private fun goIntro(context: Context, intent: Intent) {
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        intent.setClass(context, MainActivity::class.java)
        context.startActivity(intent)
    }

    private fun goMain(context: Context, intent: Intent) {
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP)
        intent.setClass(context, MainActivity::class.java)
        context.startActivity(intent)
    }
}
